lib/evmone/eof.cpp

```diff
@@ -566,14 +578,18 @@ std::variant<EOF1Header, EOFValidationError> validate_eof1(
 
         auto msh_or_error =
             validate_max_stack_height(header.get_code(container, code_idx), code_idx, header.types);
+
         if (const auto* error = std::get_if<EOFValidationError>(&msh_or_error))
             return *error;
         if (std::get<int32_t>(msh_or_error) != header.types[code_idx].max_stack_height)
             return EOFValidationError::invalid_max_stack_height;
     }
 
-    if (accessed_code_sections.size() != header.code_sizes.size())
-        return EOFValidationError::unreachable_code_sections;
+    for (auto&& visited_code_section : visited_code_sections)
```

Comment:
Why not just `const auto&`?

Reply:
can be just `const auto`, because it's a simple type, but also could use `std::all_of` or `std::any_of` instead of the loop.

Reply:
I changed it to avoid the loop, please take a look.